### DISCUSSION TITLE
Idea - Would multithreading help?

### DISCUSSION THREAD

**Discussion by ageekhere** [ThioJoe/YT-Spammer-Purge#37] [Ideas]
Would multithreading help? or is the bottleneck the Google API?

Maybe for the delete message function the work could be split into threads.

amountOfWorkPerCore = NumberOfcores/numberOfMessages (there could be a rounding error so the last thread may need to account for that)
loop through NumberOfcores
thead deleteMessage(give range to delete)

However i am not sure if the bottleneck is the Google API.

Anyway just some brainstorming

**Comment by http403** [comment]
I believe the most time spend is waiting the API to return a response. I think async will be better in this case.

**Comment by SevastianGit** [comment]
I think that it would only help if a self-hosted version of the software would be made? #5
